lib/features/digital_guide_view/general_info/presentation/digital_guide_view.dart

```diff
@@ -73,13 +73,18 @@ class _DigitalGuideView extends ConsumerWidget {
                 .replaceAll("ulica", "ul."),
             icon: Assets.svg.contactIcons.compass,
           ),
-          ContactIconsModel(
-            text: digitalGuideResponseExtended.telephoneNumber,
-            icon: Assets.svg.contactIcons.phone,
-            // TODO(Bartosh): url not working, nothing happens
-            url:
-                "tel:+48${digitalGuideResponseExtended.telephoneNumber.replaceAll("<p>", "").replaceAll("</p>", "")}",
-          ),
+          if (digitalGuideResponseExtended.telephoneNumber.isNotEmpty)
+            ContactIconsModel(
+              text: digitalGuideResponseExtended.telephoneNumber,
+              icon: Assets.svg.contactIcons.phone,
+              url: () {
+                final match = RegExp(r"\d{9}").firstMatch(
+                  digitalGuideResponseExtended.telephoneNumber
+                      .replaceAll("+48", ""),
+                );
+                return match != null ? "tel:+48${match.group(0)}" : "";
+              }(),
```

Comment:
We can move that to repository. Data that comes from it should be ready for use in ui without any modifications like that.

Reply:
Actually, I am not so sure about this. It would be advisable to use a phone number passed by API. Sometimes it return 2 phone numbers with on in parentheses. For instance: "telephone_number": "<p>713202721 (+48 665 291 063)</p>". With moving this logic to repository we wouldn't display the second number

Reply:
However, this output is random. Sometimes this phone number is formatted with spaces and sometimes not. It would be good to generalize it but then we wouldn't return 2 numbers, so idk

Reply:
Make the model class has list instead of single field